Vikta-API-testing-sample/src/test/java/com/test/api/user/UserTest.java

```diff
@@ -0,0 +1,446 @@
+        newUser.setEmail("j@k");
+        newUser.setFirstName("Jan");
+        newUser.setMiddleName("Janusz");
+        newUser.setSurname("Kowalski");
+        newUser.setPathToAvatarImage("path");
+        newUser.setAddressIds(new int[]{0});
+        newUser.setPaymentCardIds(new int[]{0});
+
+        UserDTO createdUser = given().spec(defaultRequestSpec())
+                .body(newUser)
+                .log().all()
+                .when()
+                .post(getTestEnvironment().getUserPath())
+                .then()
+                .log().all()
+                .spec(defaultResponseSpec())
+                .extract().as(UserDTO.class);
+
+        int id = createdUser.getId();
+
+        Response responseDelete = given().spec(defaultRequestSpec())
+                .queryParam(idQueryParam, id)
+                .when()
+                .log().all()
+                .delete(getTestEnvironment().getUserPath())
+                .then()
+                .log().all()
+                .spec(defaultResponseSpec())
+                .extract().response();
+
+        Response responseGetDeleted = getResponseFromUserById(id);
+
+        assertThat(responseDelete.getStatusCode()).isEqualTo(200);
+
+        assertThat(responseGetDeleted.getStatusCode())
+                .as("Category should be deleted")
+                .isEqualTo(404);
+    }
+
+    @Test
+    public void canNotDeleteNonExistingUser(){
+        int id = 0;
+
+        Response response = getResponseFromUserById(id);
+
+        assertThat(response.getStatusCode()).isEqualTo(404);
+    }
+
+    /**
+     * Method returns User and expects that user exist
+     *
+     * @param id category identifier
+     * @return UserDTO object
+     */
+    private Response getResponseFromUserById(int id){
+        return given().spec(defaultRequestSpec())
+                .queryParam(idQueryParam, id)
+                .when()
+                .get(getTestEnvironment().getUserPath())
+                .then()
+                .spec(defaultResponseSpec())
+                .extract().response();
+    }
+
+    /**
+     * Method returns List<UserDTO> and expects that user exist
+     */
+    private List<UserDTO> getListOfUsers(){
+        return given().spec(defaultRequestSpec())
+                .when()
+                .get(getTestEnvironment().getUsersListPath())
+                .then()
+                .spec(defaultResponseSpec())
+                .extract()
+                .body().jsonPath().getList(".", UserDTO.class);
+    }
+
+    private static Stream<Arguments> goodParametersForPost() {
+        String words64 = RandomStringUtils.randomAlphabetic(64);
+        String words63 = RandomStringUtils.randomAlphabetic(63);
+        String words3 = RandomStringUtils.randomAlphabetic(59);
+        // this is max that I can apply, 319 length string, more won't pass
+        String maxEmail = String.format("%s@%s.%s.%s.%s.com", words64, words63, words63, words63, words3);
```

Comment:
do we need that many string parts? maybe create one longer string for username and then one shorter for domain name?

Reply:
I checked that between dots can be max 63 chars(64 including dot), that's why here is so many string parts, but I can just check smaller string

Reply:
I think that checking against the most common email address pattern will be enough